scripts/devicetypes.py

```diff
@@ -0,0 +1,405 @@
+
+"""
+DeviceType Library Management:
+Allows:
+
+- Import DeviceTypes Library from repo
+- Update repo (via pull request)
+- Import by Manufacturer
+- Import by Model (filename)
+
+Inspired by https://github.com/netbox-community/reports/pull/34
+Requires: GitPython
+
+Installation:
+1. Copy to the scripts dir
+2. Set up parent dir (user must have write permissions within this dir)
+3. `pip install GitPython`
+"""
+
+import os
+import yaml
+from git import Repo
+
+from django.db import transaction
+from django.db.utils import IntegrityError
+from django.utils.text import slugify
+
+from dcim.choices import DeviceStatusChoices, SiteStatusChoices
+from dcim.models import (DeviceType, Manufacturer, ConsolePortTemplate,
+                         PowerPortTemplate, PowerOutletTemplate,
+                         InterfaceTemplate)
+from extras.scripts import *
+
+devicetype_parent_dir = "/opt"
+devicetype_repo_dir = "devicetype-library"
+devicetype_lib_dir = "device-types"
+
+devicetype_repo = "https://github.com/netbox-community/devicetype-library"
+
+
+class CreateDeviceType(Script):
+    
+    class Meta:
+        name = "Load devices types"
+        description = "Creates/pulls repo from github"
+
+    devicetype_parent_dir = devicetype_parent_dir
+    devicetype_repo = devicetype_repo
+
+
+    def run(self, data, commit):
+
+        devicetype_dir = os.path.join(
+            self.devicetype_parent_dir,
+            "devicetype-library"
+        )
+
+        if not os.path.isdir(self.devicetype_parent_dir):
+            self.log_error("{self.devicetype_parent_dir} is not a dir")
```

Comment:
Should be **log_failure** instead of **log_error**
`self.log_failure("{self.devicetype_parent_dir} is not a dir")`

Reply:
Thank you! I was writing it quite quickly and missed that point